packages/toolboxes/src/evm/toolbox/index.ts

```diff
@@ -1,37 +1,31 @@
 import { Chain, type EVMChain } from "@swapkit/helpers";
-import type { BrowserProvider, JsonRpcProvider } from "ethers";
 
 import { getProvider } from "../helpers";
-import type { EVMToolboxParams, EVMToolboxes } from "../types";
+import type { EVMToolboxParams } from "../types";
 import { ARBToolbox, AVAXToolbox, BASEToolbox, BSCToolbox, ETHToolbox, MATICToolbox } from "./evm";
 import { OPToolbox } from "./op";
 
-export async function getEvmToolbox<T extends EVMChain>(
-  chain: T,
-  params?: Omit<EVMToolboxParams, "provider"> & {
-    provider?: BrowserProvider | JsonRpcProvider;
-  },
-) {
+export async function getEvmToolbox<T extends EVMChain>(chain: T, params?: EVMToolboxParams) {
   const toolboxParams = {
     ...params,
     provider: params?.provider || (await getProvider(chain)),
   };
 
   switch (chain) {
     case Chain.Avalanche:
-      return AVAXToolbox(toolboxParams) as EVMToolboxes[T];
+      return AVAXToolbox(toolboxParams);
     case Chain.Arbitrum:
-      return ARBToolbox(toolboxParams) as EVMToolboxes[T];
+      return ARBToolbox(toolboxParams);
     case Chain.Base:
-      return BASEToolbox(toolboxParams) as EVMToolboxes[T];
+      return BASEToolbox(toolboxParams);
     case Chain.Optimism:
-      return OPToolbox(toolboxParams) as EVMToolboxes[T];
+      return OPToolbox(toolboxParams);
     case Chain.Polygon:
-      return MATICToolbox(toolboxParams) as EVMToolboxes[T];
+      return MATICToolbox(toolboxParams);
     case Chain.BinanceSmartChain:
-      return BSCToolbox(toolboxParams) as EVMToolboxes[T];
+      return BSCToolbox(toolboxParams);
     case Chain.Ethereum:
-      return ETHToolbox(toolboxParams) as EVMToolboxes[T];
+      return ETHToolbox(toolboxParams);
     default:
       throw new Error(`Chain ${chain} is not supported`);
   }
```

Comment:
Does it work when it comes to typing? For example if you pass one chain - does it infer proper toolbox?